### DISCUSSION TITLE
New collection review: netapp.storagegrid

### DISCUSSION THREAD

**Discussion by joshedmonds** [ansible-collections/ansible-inclusion#23] [Resolved reviews]
As per previous inclusion requests from the NetApp team, this is the last remaining NetApp collection we're requesting for inclusion.

Ansible Galaxy: https://galaxy.ansible.com/netapp/storagegrid
Github: https://github.com/ansible-collections/netapp.storagegrid
Issues Tracker: https://github.com/ansible-collections/netapp.storagegrid/issues
Github Handles: @joshedmonds @jkandati @carchi8py

**Comment by tadeboro** [comment]
Thank you for your submission, @joshedmonds.

The items in the checklist can be in one of the following states:

  * Items with a checkmark and no attached comment are OK and need no intervention from your side.
  * Items with a checkmark that contain a comment will not block the inclusion process, but by following the advice given in the comment, the collection's quality can be improved.
  * Items with no checkmark and no extra details are waiting for evaluation.
  * Items with no checkmark that contain a comment need your attention.

Feel free to ask any questions and provide feedback. Thanks in advance for your understanding and patience.

_For details about the following points, refer to the [Collection Requirements](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst)._

Every comment should say whether the reviewer expects it to be addressed, or whether it's optional.

**Public availability and communication:**
- [x] published on Ansible Galaxy [Ansible Galaxy](https://galaxy.ansible.com) with version 1.0.0 or later
- [x] has a policy of releasing, versioning and deprecation announced to contributors and users in some way
- [x] have a Code of Conduct (CoC) compatible with the [Ansible Code of Conduct (CoC)](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html)
- [x] have a publicly available issue tracker that does not require a paid level of service to create an account or view issues
- [x] have a public SCM repository, releases of the collection are tagged in this repository

**Standards and documentation:**
- [x] adheres to [semantic versioning](https://semver.org/)
- [x] follows [licensing rules](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#licensing)
- [x] follows the [Ansible documentation standards](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html) and the [style guide](https://docs.ansible.com/ansible/devel/dev_guide/style_guide/index.html#style-guide)
- [x] follows [development conventions](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_best_practices.html); as well as these other requirements:
  - [x] modules that only gather information are named `<something>_info`
  - [x] modules that return `ansible_facts` are named `<something>_facts` and do not return non-facts
  - [x] other modules must not allow querying information using specific `state` option values, or similar mechanisms (like `state=get` or `state=query`).  These features should be moved to `<something>_info` or `<something>_fact` modules.
  - [x] `check_mode` is supported in all `_info` and `_facts` modules
- [x] supports Python 2.6 or greater and Python 3.5 or greater. If it does not, read the [full guidelines](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#python-compatibility) to see if you qualify for an exception and document the unsupported [Python versions](https://docs.ansible.com/ansible/latest/dev_guide/developing_python_3.html#ansible-and-python-3) in the collection ``README.md`` and in every module and plugin (or in doc fragments)
- [x] only uses the [allowed plugin types](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#modules-plugins) in the `plugins/` directory
- [x] has `README.md`
- [x] documentation, examples, and return sections use FQCNs for the `M(..)` [format macros](https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#linking-and-other-format-macros-within-module-documentation) when referring to modules
- [x] modules (or plugins) from ansible-core use `ansible.builtin.` as a FQCN prefix
- [x] documentation and return sections use `version_added:` containing the *collection* version for which an option, module or plugin was added
- [x] FQCNs are used for `extends_documentation_fragment:`, unless you are referring to doc_fragments from ansible-core

**Collection management:**
- [x] `galaxy.yml` having `tags` field set
- [x] collection dependencies must have a lower bound on the version which is at least 1.0.0, and are all part of the `ansible` package
- [x] `meta/runtime.yml` defines the minimal version of Ansible which the collection works with
- [x] has changelog, preferably with `changelogs/changelog.yaml`

**Tests:**
- [x] passed `ansible-test sanity`
- [x] if `test/sanity/ignore*.txt` exists, it does not contain error codes listed [here](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#ci-testing)
- [x] has CI tests up and running against each of the "major versions" of `ansible-base`/`ansible-core` that the collection supports
- [x] all CI tests run against every pull request
- [x] all CI tests run against a commit that releases the collection; if they don't pass, the collection won't be released
- [x] all CI tests run regularly (nightly, or at least once per week)

**Reply by tadeboro** [reply]
I finished my initial review. The collection cannot be included in the `ansible` package right now.

**Reply by joshedmonds** [reply]
Thanks for the review. We are in the process of addressing each of the outstanding items.

**Reply by joshedmonds** [reply]
@tadeboro we believe that all of the outstanding items have now been addressed. Note, the readme file points to the release and versioning policy that is common with all of the other netapp.* collections.

**Reply by tadeboro** [reply]
I will do another round of reviews in about a week. Right now, I am a bit swamped with other work. Thank you for keeping things moving.

**Reply by tadeboro** [reply]
I did a second review, and the collection is ready for inclusion as far as I am concerned. All it needs is another release on Ansible Galaxy with the changes from the main branch. Thank you, @joshedmonds, for your effort!

**Reply by joshedmonds** [reply]
@tadeboro I've now published 21.7.0 on Ansible Galaxy which includes all of the above.

**Comment by jillr** [comment]
Collection LGTM, +1 for inclusion. Thanks @joshedmonds!

**Comment by felixfontein** [comment]
While looking at the collection, I found a few docs problems:
1. In many places, using `C(...)` or `I(...)` would improve the documentation (see https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#linking-and-other-format-macros-within-module-documentation).
2. The texts in `description:` should consist of full sentences. There are a lot of them where a final period is missing.
3. There's a `\` in https://github.com/ansible-collections/netapp.storagegrid/blob/main/plugins/doc_fragments/netapp.py#L40 which doesn't belong there I think?
4. Some module examples have no FQCN, see for example https://github.com/ansible-collections/netapp.storagegrid/blob/main/plugins/modules/na_sg_grid_info.py#L80 and https://github.com/ansible-collections/netapp.storagegrid/blob/main/plugins/modules/na_sg_org_info.py#L66

**Reply by joshedmonds** [reply]
Thanks @felixfontein. Will resolve these ASAP.

**Reply by joshedmonds** [reply]
@felixfontein I've addressed these documentation issues in the latest commit: https://github.com/ansible-collections/netapp.storagegrid/commit/d691b7dc5c6a61c0a6b20fe304318750c4f494fb

**Reply by felixfontein** [reply]
@joshedmonds thanks, that looks good!

**Comment by tadeboro** [comment]
The community members accepted the `netapp.storagegrid` collection into Ansible 5 (https://github.com/ansible-community/ansible-build-data/pull/73). Thank you all for your patience and cooperation.
